src/server/ua_services_discovery.c

```diff
@@ -118,3 +232,65 @@ void Service_GetEndpoints(UA_Server *server, UA_Session *session, const UA_GetEn
         return;
     }
 }
+
+
+void Service_RegisterServer(UA_Server *server, UA_Session *session,
+                         const UA_RegisterServerRequest *request, UA_RegisterServerResponse *response) {
+    UA_LOG_DEBUG_SESSION(server->config.logger, session, "Processing RegisterServerRequest");
+
+    registeredServer_list_entry *registeredServer_entry = NULL;
+
+    {
```

Comment:
ANSI C (89) required variables to be declared at the top of a block.
With C99 that is no longer the case. And most compilers allowed it anyway.

The variable is already "hidden" as it is local to that function.
So the bock seems mostly cosmetic to me.